loleaflet/src/layer/tile/CanvasSectionContainer.ts

```diff
@@ -285,10 +311,205 @@ class CanvasSectionContainer {
 		}
 	}
 
+	private updateBoundSectionList(section: CanvasSectionObject, sectionList: Array<CanvasSectionObject> = null): Array<CanvasSectionObject>{
+		if (sectionList === null)
+			sectionList = new Array(0);
+
+		sectionList.push(section);
+
+		var tempSectionList: Array<CanvasSectionObject> = new Array(0);
+
+		if (section.boundToSection) {
+			var tempSection = this.getSectionWithName(section.boundToSection);
+			if (tempSection && tempSection.isLocated) {
+				if (tempSection.zIndex < section.zIndex || (tempSection.zIndex === section.zIndex && tempSection.drawingOrder < section.drawingOrder)) {
+					tempSectionList.push(tempSection);
+				}
+			}
+		}
+
+		for (var i: number = 0; i < this.sections.length; i++) {
+			if (this.sections[i].isLocated && this.sections[i].boundToSection === section.name) {
+				if (this.sections[i].zIndex < section.zIndex || (this.sections[i].zIndex === section.zIndex && this.sections[i].drawingOrder < section.drawingOrder)) {
+					tempSectionList.push(this.sections[i]);
+				}
+			}
+		}
+
+		for (var i: number = 0; i < tempSectionList.length; i++) {
+			this.updateBoundSectionList(tempSectionList[i], sectionList);
+		}
+
+		return sectionList;
+	}
+
+	private orderBoundsList(section: CanvasSectionObject) {
+		// According to zIndex.
+		for (var i: number = 0; i < section.boundsList.length - 1; i++) {
+			for (var j = i + 1; j < section.boundsList.length; j++) {
```

Comment:
These two can be done together in a double for loop (or better using a comparison function passed to array builtin sort()) so that it is easier to understand and yet shorter code in my opinion.

Reply:
İf you mean two ordering parts may be merged, i think it might cause buggy behaviour. Second one relies on the first's results. For comparison function, this is a small bits of code. Do we real need that:)

Thank you for the comments and your time.

Reply:
It might seem so, but it is equivalent to a single sort with a comparison function (or its reverse depending on how you want to sort) :

if (one.zIndex > two.zIndex || (one.zIndex == two.zIndex && one.drawingOrder > two.drawingOrder) )
   return -1; // one less than two, so in ascending sort(), one is placed before two.
if (one.zIndex == two.zIndex && one.drawingOrder == two.drawingOrder)
  return 0; // equal (no swap)
return 1; // one greater than two.

(The second double-for is basically moving around just the ones that have same zindex.)

Reply:
a single sort() with comparison function will save the trouble of writing a double for loop every time we want to sort :)

Reply:
Oh, this is a smart solution:) Thanks.